### PR TITLE
feat: rate limit requests to target registry

### LINKED ISSUE #20: Reduce refresh pull rate
The registry pulling rate when refreshing staticreg content should be reduced to prevent polluting access metrics. 

Slack convo [here](https://seqera.slack.com/archives/C03117E4RAB/p1729516465386479).

### PR DESCRIPTION
**Pull request by fntlnz**
Fixes #20

### CODE REVIEW

**Review by alberto-miranda — APPROVED** [review]
LGTM! 💃

File: cmd/serve.go [hunk 1]
```diff
@@ -93,5 +99,6 @@ func init() {
 	serveCmd.PersistentFlags().StringArrayVar(&ignoredUserAgents, "ignored-user-agent", []string{}, "user agents to ignore (reply with empty body and 200 OK). A user agent is ignored if it contains the one of the values passed to this flag")
 	serveCmd.PersistentFlags().DurationVar(&cacheDuration, "cache-duration", time.Minute*1, "how long to keep a generated page in cache before expiring it, 0 to never expire")
 	serveCmd.PersistentFlags().DurationVar(&refreshInterval, "refresh-interval", time.Minute*15, "how long to wait before trying to get fresh data from the target registry")
+	serveCmd.PersistentFlags().IntVar(&requestsPerSecond, "requests-per-second", 1, "limit the number of requests per second that can be done to the target registry")
```
Comment (jordeu) [inline cmd/serve.go:102]: Is one per second a good default? It seems a very low rate.

**Review by jordeu — APPROVED** [review]
LGTM, maybe I'd consider using [this](https://failsafe-go.dev/rate-limiter/) rate limiter on future improvements if needed. It has nice features, like being able to log an entry when the rate limit is hit.

File: cmd/serve.go (hunk 1 above)
Comment (fntlnz) [reply cmd/serve.go:102]: yes it's low but I tested it locally against wave community and it feels fast enough. The async registry is updating the internal registry representatation in background so it's an issue visible only on restart.

Comment (fntlnz) [reply cmd/serve.go:102]: Agreed that rate limiter is one I also like, felt like it was easy enough to avoid a dependency however.

**Pull request merged**